### PR TITLE
fix: import whatwg-url in a way compatible with ESM Node

### LINKED ISSUE #1305: node-fetch@2.6.4 build error in webpack@4, node-fetch@2.6.2 works fine
**Reproduction**
- https://unpkg.alipay.com/browse/node-fetch@2.6.4/lib/index.mjs
-  `import { URL } from 'whatwg-url';`  （es module）
-  https://unpkg.alipay.com/browse/whatwg-url@5.0.0/lib/public-api.js  (commonjs, no default export)
-  `ModuleDependencyError: Can't import the named export 'URL' from non EcmaScript module (only default export is available)`

**Expected behavior**

**Screenshots**

**Your Environment**

| software         | version
| ---------------- | -------
| node-fetch       |
| node             |
| npm              |
| Operating System |

**Additional context**

### PR DESCRIPTION
**Pull request by LinusU**
**What is the purpose of this pull request?**

- Documentation update
- [x] Bug fix
- New feature
- Other, please explain:

**What changes did you make? (provide an overview)**

I imported `whatwg-url` in a way that is compatible with Node.js running in ESM mode.

**Which issue (if any) does this pull request address?**

https://github.com/node-fetch/node-fetch/commit/ace7536c955556be742d9910566738630cc3c2a6#r56836783

**Is there anything you'd like reviewers to know?**

n/a

closes #1305

### CODE REVIEW

File: src/request.js [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 import Url from 'url';
 import Stream from 'stream';
-import {URL} from 'whatwg-url';
+import whatwgUrl from 'whatwg-url';
 import Headers, { exportNodeCompatibleHeaders } from './headers.js';
 import Body, { clone, extractContentType, getTotalBytes } from './body';
```
Comment (jimmywarting) [inline src/request.js:15]: maybe do: `const URL = whatwgUrl.URL` then?

so it can GC the rest

**Comment by jimmywarting** [comment]
Any possible way to test this also?
changelog + patch version also?

**Comment by LinusU** [comment]
> Any possible way to unit test this also?

I haven't found one yet, since I cannot trigger the issue myself (even though I think that you shouldn't be able to import it like that 😄)

Waiting for more information from @robrecord

> changelog + patch version also?

Will fix ✅

**Comment by robrecord** [comment]
I have confirmed this fixes the error I was facing. Thank you!

**Comment by paranoidjk** [comment]
This should fix  https://github.com/node-fetch/node-fetch/issues/1305

**Comment by LinusU** [comment]
@robrecord or @paranoidjk do you know of any easy way we could add a test for this?

**Comment by LinusU** [comment]
@jimmywarting I've added changelog entry and bumped the package version. What do you think of merging this without a test for now, and adding one later as this seems to be breaking some peoples workflow and they have to pin node-fetch at a lower version currently?

**Comment by jimmywarting** [comment]
Yea, can release it without a test

**Review by jimmywarting — APPROVED** [review]

**Comment by robrecord** [comment]
If I can find an easy way to have this tested I will add it here.

**Comment by jimmywarting** [comment]
> If I can find an easy way to have this tested I will add it here.

Grate

**Pull request merged**